lib/SIL/SILDeclRef.cpp

```diff
@@ -888,12 +865,23 @@ SubclassScope SILDeclRef::getSubclassScope() const {
   llvm_unreachable("Unhandled access level in switch.");
 }
 
-unsigned SILDeclRef::getUncurryLevel() const {
-  if (isCurried)
-    return 0;
-  if (!hasDecl())
-    return getFuncNaturalUncurryLevel(*getAnyFunctionRef());
-  if (kind == Kind::DefaultArgGenerator)
-    return 0;
-  return getNaturalUncurryLevel(getDecl());
+unsigned SILDeclRef::getParameterListCount() const {
```

Comment:
Can you add more documentation here and use early-exits instead of else-if return?

What I want is for each case to say /why/ it has X amounts of parameters. For instance, why does ClassDecl have 2 parameters? It would also be nice to have a bigger comment explaining this methods use/importance in the larger world of SIL.